shared/actions/kbfs/index.platform.desktop.js

```diff
@@ -164,63 +164,57 @@ function* installFuseSaga(): SagaGenerator<any, any> {
-
-        for (var keyName of programKeyNames) {
-          var programKey = 'HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\' + keyName
-
-          regedit.list(programKey, (err, program) => {
-            if (err) {
-              reject(err)
-            } else {
-              for (var p in program) {
-                var vals = program[p]
-                var displayName, publisher
-                var modifyPath = ''
-                for (var v in vals) {
-                  if (vals[v]['DisplayName']) {
-                    displayName = vals[v]['DisplayName'].value
-                  }
-                  if (vals[v]['Publisher']) {
-                    publisher = vals[v]['Publisher'].value
-                  }
-                  if (vals[v]['ModifyPath']) {
-                    modifyPath = vals[v]['ModifyPath'].value
-                  }
-                }
-                if (displayName === 'Keybase' && publisher === 'Keybase, Inc.') {
-                  // Remove double quotes - won't work otherwise
-                  modifyPath = modifyPath.replace(/"/g, '')
-                  // Remove /modify and send it in with the other arguments, below
-                  modifyPath = modifyPath.replace(' /modify', '')
-                  console.log(modifyPath)
-                  execFile(modifyPath, [
-                    '/modify',
-                    'driver=1',
-                    'modifyprompt=Press "Repair" to view files in Explorer',
-                  ])
-                  resolve()
-                }
-              }
+    regedit.list('HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall').on('data', function(entry) {
+      for (var keyName of entry.data.keys) {
+        regedit
+          .list('HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\' + keyName)
+          .on('data', function(entry) {
+            if (
+              entry.data.values.DisplayName &&
+              entry.data.values.DisplayName.value === 'Keybase' &&
+              entry.data.values.Publisher &&
+              entry.data.values.Publisher.value === 'Keybase, Inc.' &&
+              entry.data.values.ModifyPath
+            ) {
+              var modifyPath = entry.data.values.ModifyPath.value
+              // Remove double quotes - won't work otherwise
+              modifyPath = modifyPath.replace(/"/g, '')
+              // Remove /modify and send it in with the other arguments, below
+              modifyPath = modifyPath.replace(' /modify', '')
+              resolve(modifyPath)
             }
           })
-        }
       }
     })
   })
 }
 
+// Invoking the cached installer package has to happen from the topmost process
+// or it won't be visible to the user. The service also does this to support command line
+// operations.
+function installCachedDokan(): Promise<*> {
+  return new Promise(async (resolve, reject) => {
```

Comment:
we don't use async anywhere else, not sure we should use it here.

Reply:
rewritten would look something like:
```js
function installCachedDokan(): Promise<void> {
  return findKeybaseUninstallString().then((modifyCommand) => new Promise (resolve, reject) => {
    // ... same as below
  })
}
```

Reply:
I agree, but I think we should also discuss if we want to start using it. It matches stylistically what we're doing with redux-saga so it might be nice to start using it. lets chat on slack

Reply:
Whatever you guys say, I'm still learning all this